app/scripts/metamask-controller.js

```diff
@@ -1311,19 +1310,11 @@ export default class MetamaskController extends EventEmitter {
     );
     this.smartTransactionsController = new SmartTransactionsController(
       {
-        onNetworkStateChange: (cb) => {
-          this.networkController.store.subscribe((networkState) => {
-            const modifiedNetworkState = {
-              ...networkState,
-              providerConfig: {
-                ...networkState.providerConfig,
-              },
-            };
-            return cb(modifiedNetworkState);
-          });
-        },
-        getNetwork: () =>
-          this.networkController.store.getState().networkId ?? 'loading',
+        onNetworkStateChange: networkControllerMessenger.subscribe.bind(
```

Comment:
It seemed that previously we were copying the state but making no other changes to the state. I've replaced this with a direct bind to the `stateChange` event.

Reply:
Looks like this was originally done to work around the need to rename `provider` to `providerConfig` (see https://github.com/MetaMask/metamask-extension/commit/e0ff039078e7ffd54c464aa856de13b41d30ea9a), but I must have missed that it was no longer necessary after that was finally renamed in https://github.com/MetaMask/metamask-extension/commit/d1cea85f3360801e9753d5ef126fd49797cd1ffe#diff-6fbff2cfe97ac01b77296ef2122c7e0a5b3ff6a84b584b4d1a87482f35eea3d6